### DISCUSSION TITLE
2d渲染模式和3d渲染模式的区别是什么?

### DISCUSSION THREAD

**Discussion by rerender2021** [qber-soft/Ave-Nodejs#43] [Q&A] [answered]
默认是2d，这样是3d：

```bash
cpWindow.Device = WindowDevice.Default3D;
```

区别是什么呢。。除了可以使用3d ui之外

**Comment by ivjtk** [comment, marked as answer]
定制的3D渲染器主要就是性能高，大约是2D(D2D)的20~100倍以上（根据GPU有所区别）。

缺点就是现在实现不完整，不支持ClearType字体渲染，有些特殊字符不支持，比如高棉语、希伯来语等（表现为字符显示位置或形状不正确），阿拉伯语是支持的（右到左和变形都支持），另外竖排文本也没有完全实现，主要就是体现在文字渲染上功能还没有实现好。
